src/typecheck.rs

```diff
@@ -1742,6 +1768,25 @@ pub fn get_bop_type(state: &mut State, op: &BinaryOp) -> Result<TypeWrapper, Typ
             mk_typewrapper::num(),
             mk_typewrapper::num()
         ),
+        // Str -> Str -> Bool
+        BinaryOp::StrContains() => mk_tyw_arrow!(AbsType::Str(), AbsType::Str(), AbsType::Bool()),
```

Comment:
It's painful for the primitive operator `StrContains`, because it needs fully evaluated arguments. However, this operator isn't supposed to be used directly by users and is wrapped in the stdlib: we can consider having the user-facing function `strings.contains` to take a record indeed. But I think this belongs to a more general stdlib design discussion about named arguments (if we start there, then a lot of other functions of the stdlib should also have named arguments), which also have drawbacks (partial application becomes harder, for example). I propose to open a separate issue and let `contains` as it is in this PR for now.